File: src/main/java/idlezoo/security/IdUser.java
package idlezoo.security;

import java.util.Collections;

import org.springframework.security.core.userdetails.User;

public class IdUser extends User {

    private static final long serialVersionUID = 1L;
    private final Integer id;

    public IdUser(Integer id, String username, String password) {
        super(username, password, Collections.emptyList());
        this.id = id;
    }

    public Integer getId() {
        return id;
    }

}

File: src/main/kotlin/idlezoo/security/IdUser.kt
package idlezoo.security

import org.springframework.security.core.GrantedAuthority
import java.util.Collections

import org.springframework.security.core.userdetails.User

class IdUser(val id: Int?, username: String, password: String) : User(username, password, emptyList<GrantedAuthority>())
